platforms/web/web.js

```diff
@@ -0,0 +1,181 @@
+    if (typeof uResponse === "string") {
+      return getTextMessage(uResponse);
+    }
+
+    if (uResponse instanceof TextMessage) {
+      return getTextMessage(uResponse.text);
+    }
+
+    if (uResponse instanceof Button) {
+      return getButton(uResponse);
+    }
+
+    if (uResponse instanceof ButtonsListMessage || uResponse instanceof ButtonsMessage) {
+      return getListButton(uResponse);
+    }
+
+    return uResponse;
+  });
+
+}
+
+function sendFeedback(senderId, intent, message) {
+  if (intent === "unknown") {
+    return;
+  }
+
+  message = message.length >= 1000 ? "TOOLONG" : message;
+
+  let buttons = [
+    new Button("postback", `FEEDBACK_MISUNDERSTOOD_${camelCase(intent)}_${message}`, "Mauvaise compréhension"),
+    new Button("postback", `FEEDBACK_BAD_${camelCase(intent)}_${message}`, "Non"),
+    new Button("postback", `FEEDBACK_GOOD_${camelCase(intent)}_${message}`, "Oui")
+  ];
+
+  return send(null, senderId, new ButtonsListMessage("Est-ce que cette réponse vous a aidé ?", buttons));
+}
+
+
+
+function send(res, id, rawResponses, opt) {
+  let flagUseRes = true;
+  let senderId = id;
+  let responses;
+
+  //input validation
+  if (!rawResponses) {
+    rawResponses = id;
+    senderId = null;
+  }
+
+  //detect if res is the senderid or the response from express.
+  if (typeof res === "string") {
+    senderId = res;
+    flagUseRes = false;
+  }
+
+  //check if the response was used
+  if (!res || (res && res.headersSent)) {
+    flagUseRes = false;
+  }
+
+  //validate the senderId
+  if (typeof senderId !== "string") {
+    senderId = null;
+  }
+
+  //parse the rawResponses into a suitable format
+  responses = parseMsg(rawResponses);
+
+  //push the message to db for conversation history
+  if (responses.length > 0) {
+    return Bluebird.map(responses, (historyMessage) => {
+      historyMessage.origin = "bot";
+      return pushToHistory(senderId, historyMessage);
+    })
+    .then(() => {
+      //if we cant answer, we save to db for later
+      if (flagUseRes) {
+        return Bluebird.resolve(responses)
+          .then((result) => {
+            if (opt && opt.feedback) {
+              sendFeedback(senderId, opt.intent, opt.message);
+            }
+            return result;
+          });
+      } else if (senderId) {
+        return WebMessage.findOneAndUpdate({senderId}, {$push: {unread: {$each: responses}}}, {upsert: true})
+           .exec()
+           .then((result) => {
+             if (opt && opt.feedback) {
+               sendFeedback(senderId, opt.intent, opt.message);
+             }
+             return result;
+           });
+      } else {
+        return Bluebird.reject(new Error("Il semblerait qu'il n'y a pas de canal de retour pour ce message :" + responses));
+      }
+    });
+  }
+
+}
+
+
+function getUnread(res, senderId) {
+  return WebMessage
+    .findOneAndUpdate({senderId}, {unread: []}, {upsert: true})
+    .lean()
+    .exec()
+    .then(result => {
```

Comment:
I would not split `.then` and `.catch` in different parts of the code.

Either you `.then().catch()` in `web.js`, and return the result.
Or you might want to return the whole WebMessage object in `getUnread` (until exec()), and then `then().catch()` in the controller.